### PR TITLE
mgmt, treat subscription id as string

### PR DESCRIPTION
**Pull request by XiaofeiCao**
fix https://github.com/Azure/autorest.java/issues/1976

Solution is to map parameter type to String in both ManagementClient and proxy methods.

### CODE REVIEW

File: fluent-tests/prepare-tests.bat [hunk 1]
```diff
@@ -83,6 +87,10 @@ if %errorlevel% neq 0 exit /b %errorlevel%
 CALL autorest --version=%AUTOREST_CORE_VERSION% %FLUENTLITE_ARGUMENTS% --regenerate-pom=false https://raw.githubusercontent.com/Azure/azure-rest-api-specs/main/specification/resources/resource-manager/readme.md --tag=package-policy-2020-09 --java.namespace=com.azure.mgmtlitetest.policy
 if %errorlevel% neq 0 exit /b %errorlevel%
 
+REM fluent lite uuid subscriptionId
+CALL autorest --version=%AUTOREST_CORE_VERSION% %FLUENTLITE_ARGUMENTS% --input-file=https://github.com/Azure/azure-rest-api-specs/blob/0a2eb0d14f5132fcfd30222d584acf67713332ea/specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/stable/2022-12-01/containerregistry.json --namespace=com.azure.mgmtlitetest.containerregistrylite
+if %errorlevel% neq 0 exit /b %errorlevel%
+
```
Comment (weidongxu-microsoft) [inline fluent-tests/prepare-tests.bat:93]: nit, Lite should be enough. That should cover premium.

File: javagen/src/main/java/com/azure/autorest/mapper/ProxyParameterMapper.java [hunk 2]
```diff
@@ -58,6 +58,10 @@ public ProxyMethodParameter map(Parameter parameter) {
 
         Schema parameterJvWireType = parameter.getSchema();
         IType wireType = Mappers.getSchemaMapper().map(parameterJvWireType);
+        // fluent premium treats subscriptionId as string in order not to break ManagementClients and client methods
+        if (settings.isFluentPremium() && "subscriptionId".equals(CodeNamer.getPropertyName(name))) {
```
Comment (weidongxu-microsoft) [inline javagen/src/main/java/com/azure/autorest/mapper/ProxyParameterMapper.java:62]: why `isFluentPremium`?

**Review by weidongxu-microsoft — COMMENTED** [review]
Not sure if `FluentTransformer` be a better place? E.g. only check those under `codeModel.globalParameters`. If so, modify the schema (may need to create a StringSchema).

We don't want to match by name for all parameters. Who knows whether some one want to have a valid `subscriptionId` param in api.

File: javagen/src/main/java/com/azure/autorest/mapper/ProxyParameterMapper.java (hunk 2 above)
Comment (XiaofeiCao) [reply javagen/src/main/java/com/azure/autorest/mapper/ProxyParameterMapper.java:62]: Thought we may break existing managementClient in lite?

Comment (weidongxu-microsoft) [reply javagen/src/main/java/com/azure/autorest/mapper/ProxyParameterMapper.java:62]: It does not break. Officially ClientManagement is not included in revapi.

Comment (XiaofeiCao) [reply javagen/src/main/java/com/azure/autorest/mapper/ProxyParameterMapper.java:62]: OK. FluentTransformer should be a better place to make the change then.

**Pull request merged**